src/smartAuthorizationHelper.ts

```diff
@@ -1,42 +1,54 @@
-import { KeyValueMap, UnauthorizedError } from 'fhir-works-on-aws-interface';
-import { IdentityType } from './smartConfig';
+import { UnauthorizedError } from 'fhir-works-on-aws-interface';
 
 export const FHIR_USER_REGEX = /^(?<hostname>(http|https):\/\/([A-Za-z0-9\-\\.:%$_]*\/)+)(?<resourceType>Person|Practitioner|RelatedPerson|Patient)\/(?<id>[A-Za-z0-9\-.]+)$/;
+export const FHIR_RESOURCE_REGEX = /(?<hostname>^(http|https):\/\/([A-Za-z0-9\-\\.:%$_]*\/)+)?(?<resourceType>[A-Z][a-zA-Z]+)\/(?<id>[A-Za-z0-9\-.]+)$/;
 
-export interface FhirUser {
+export interface FhirResource {
     hostname: string;
-    resourceType: IdentityType;
+    resourceType: string;
     id: string;
 }
-export function getFhirUser(userIdentity: KeyValueMap, fhirUserClaimKey: string): FhirUser {
-    const fhirUserValue = userIdentity[fhirUserClaimKey];
+export function getFhirUser(fhirUserValue: string): FhirResource {
     const match = fhirUserValue.match(FHIR_USER_REGEX);
     if (match) {
         const { hostname, resourceType, id } = match.groups!;
         return { hostname, resourceType, id };
     }
     throw new UnauthorizedError("Requester's identity is in the incorrect format");
 }
+export function getFhirResource(resourceValue: string, defaultHostname: string): FhirResource {
+    const match = resourceValue.match(FHIR_RESOURCE_REGEX);
+    if (match) {
+        // eslint-disable-next-line prefer-const
+        let { hostname, resourceType, id } = match.groups!;
+        if (!hostname) {
+            hostname = defaultHostname;
+        }
```

Comment:
I think that we should avoid disabling eslint rules unless we have a strong reason to do so. 
```suggestion
        const { resourceType, id } = match.groups!;
        const hostname = match.groups!.hostname ?? defaultHostname;
```

Reply:
fixed